modules/electromagnetics/src/functions/WaveCoeff.C

```diff
@@ -0,0 +1,68 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "WaveCoeff.h"
+#include "ElectromagneticEnums.h"
+#include <complex>
+
+registerMooseObject("ElectromagneticsApp", WaveCoeff);
+
+InputParameters
+WaveCoeff::validParams()
+{
+  InputParameters params = Function::validParams();
+  params.addClassDescription(
+      "Function for use as coefficient in standard-form Helmholtz wave equation applications.");
+  params.addRequiredParam<FunctionName>("eps_rel_real", "Relative permittivity, real component.");
+  params.addRequiredParam<FunctionName>("eps_rel_imag",
+                                        "Relative permittivity, imaginary component.");
+  params.addRequiredParam<FunctionName>("mu_rel_real", "Relative permeability, real component.");
+  params.addRequiredParam<FunctionName>("mu_rel_imag",
+                                        "Relative permeability, imaginary component.");
```

Comment:
are these always going to be functions? Or will they be material propoerties in a multiphysics coupled simulation?

Reply:
They will eventually be material properties (and in the longer term be tensor material properties). So should I switch to functors here too?